openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -439,6 +442,21 @@ def test_course_messaging(self):
         self.assertNotContains(response, TEST_COURSE_HOME_MESSAGE_UNENROLLED)
         self.assertNotContains(response, TEST_COURSE_HOME_MESSAGE_PRE_START)
 
+        # Verify that enrolled users are shown the course expiration banner if content gating is enabled
+        with override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True):
```

Comment:
would we also want to explicitly test that the expiration messages is not visible when the flag is off?

Reply:
@mulby, question for you  
at one point we had a discussion where you said we would specifically have all our tests run with the flag on because that's what edx.org will have and the open source use case was not a priority for us at the moment
we did break things with the flag off though, so maybe we now feel otherwise?  
if we do want to write tests for both cases though, we should add some tickets to figure out how to fully cover all relevant tests with the flag both on and off.  
right now we have a bit of a hodge podge of both cases

Reply:
I created this to capture that work a while ago:

https://openedx.atlassian.net/browse/REVE-75